src/Workspaces/Core/Portable/Workspace/Solution/TextLoader.cs

```diff
@@ -29,83 +34,107 @@ public abstract class TextLoader
         /// <exception cref="IOException" />
         /// <exception cref="InvalidDataException"/>
         /// <exception cref="OperationCanceledException"/>
-        public abstract Task<TextAndVersion> LoadTextAndVersionAsync(Workspace workspace, DocumentId documentId, CancellationToken cancellationToken);
+        public virtual Task<TextAndVersion> LoadTextAndVersionAsync(CancellationToken cancellationToken)
+        {
+#pragma warning disable CS0618 // Type or member is obsolete
+            if (ImmutableInterlocked.GetOrAdd(
+                ref s_isObsoleteLoadTextAndVersionAsyncOverriden,
+                GetType(),
```

Comment:
>GetType() [](http://example.com/codeflow?start=16&length=9)

Is this going to keep types alive indefinitely, preventing collectable types from unloading?

Reply:
Yes. I'm not sure the cache adds much benefit.

Reply:
Perhaps the cache should be conditional weak table.

Reply:
Yup, changed.